elastic/kibana
# [Upgrade Assistant] Show on prem backup step when found-snapshots repository not found

### Summary
As described by [ECE-Snapshots docs](https://www.elastic.co/guide/en/cloud-enterprise/current/ece-snapshots.html) users might not have snapshots configured by default. The `cloud backup step` assumes that the installation has it and that the `found-snapshots` repository exists, as a consequence of this the user will be seeing an error about a missing snapshots repository.

This PR changes the logic of the backup step to detect this edge case and default to the on prem step that will instead render a button that will take the user to the snapshots app.


**How to test**
1. Start elasticsearch with `yarn es snapshot --license=trial` and kibana with `yarn start`
2. Change [this line](https://github.com/elastic/kibana/pull/121060/files#diff-70a408cebea90dd01b02d11ec507814b12d7c4f96da81c887f8a179210aecd59R28) so that it negates the first part of the if expression that checks for `isCloudEnabled`. This will force to load the cloud backup step while working locally.
3. Navigate to `Stack Management` -> `Upgrade Assistant`
4. Verify that the network request for `cloud_backup_status` failed and that the backupstep is for on prem

Note: when this PR is merged, I'll create a new one for getting this change into main and 8.x

**Screenshots**

<details>
<summary>Error message</summary>

![image](https://user-images.githubusercontent.com/1191206/145805774-0fa9c39b-d5f1-4f53-b8bd-967cc9fa67d5.png)



</details>
